tests/test_workers/test_pkg_managers/test_pip.py

```diff
@@ -0,0 +1,587 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import contextlib
+import logging
+from pathlib import Path
+from textwrap import dedent
+
+import pytest
+
+from cachito.workers.pkg_managers import pip
+from cachito.errors import ValidationError
+
+
+def setup_module():
+    """Re-enable logging that was disabled at some point in previous tests."""
+    pip.log.disabled = False
+    pip.log.setLevel(logging.DEBUG)
+
+
+def write_file_tree(tree_def, rooted_at):
+    """
+    Write a file tree to disk.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree, must be an existing directory
+    """
+    root = Path(rooted_at)
+    for entry, value in tree_def.items():
+        entry_path = root / entry
+        if isinstance(value, str):
+            entry_path.write_text(value)
+        else:
+            entry_path.mkdir()
+            write_file_tree(value, entry_path)
+
+
+def remove_file_tree(tree_def, rooted_at):
+    """
+    Remove a file tree from disk.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree, will not be removed
+    """
+    root = Path(rooted_at)
+    for entry, value in tree_def.items():
+        entry_path = root / entry
+        if isinstance(value, str):
+            entry_path.unlink()
+        else:
+            remove_file_tree(value, entry_path)
+            entry_path.rmdir()
+
+
+@contextlib.contextmanager
+def file_tree(tree_def, rooted_at):
+    """
+    Create a file tree for the duration of the context.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree
+    """
+    write_file_tree(tree_def, rooted_at)
+    try:
+        yield
+    finally:
+        remove_file_tree(tree_def, rooted_at)
+
+
+class TestSetupCFG:
+    """SetupCFG tests."""
+
+    @pytest.mark.parametrize("exists", [True, False])
+    def test_exists(self, exists, tmpdir):
+        """Test file existence check."""
+        if exists:
+            tmpdir.join("setup.cfg").write("")
+
+        setup_cfg = pip.SetupCFG(tmpdir.strpath)
+        assert setup_cfg.exists() == exists
+
+    @pytest.mark.parametrize(
+        "cfg_content, expect_name, expect_logs",
+        [
+            (
+                "",
+                None,
+                ["Parsing setup.cfg at '{tmpdir}/setup.cfg'", "No metadata.name in setup.cfg"],
+            ),
+            ("[metadata]", None, ["No metadata.name in setup.cfg"]),
+            (
+                dedent(
+                    """
+            [metadata]
```

Comment:
Is Black pushing for thins indentation? If so, try adding a backslash after `"""`:
```python
dedent(
  """\
  [metadata]
  ...
  """
)

Reply:
Oh, that feels *so* much better :D